.github/workflows/plgd-device-tests.yml

```diff
@@ -44,62 +44,67 @@ jobs:
+            docker_file: docker/apps/Dockerfile.cloud-server-debug-clang
 
           - name: cloud-server-rep-realloc
-            args: "-DOC_REPRESENTATION_REALLOC_ENCODING_ENABLED=ON"
+            args: "-DOC_DEBUG_ENABLED=ON -DOC_REPRESENTATION_REALLOC_ENCODING_ENABLED=ON"
           - name: cloud-server-rep-realloc-asan
-            args: "-DOC_REPRESENTATION_REALLOC_ENCODING_ENABLED=ON -DOC_ASAN_ENABLED=ON"
+            args: "-DOC_DEBUG_ENABLED=ON -DOC_REPRESENTATION_REALLOC_ENCODING_ENABLED=ON -DOC_ASAN_ENABLED=ON"
           - name: cloud-server-rep-realloc-tsan
-            args: "-DOC_REPRESENTATION_REALLOC_ENCODING_ENABLED=ON -DOC_TSAN_ENABLED=ON"
-            clang: true
+            args: "-DOC_DEBUG_ENABLED=ON -DOC_REPRESENTATION_REALLOC_ENCODING_ENABLED=ON -DOC_TSAN_ENABLED=ON"
+            docker_file: docker/apps/Dockerfile.cloud-server-debug-clang
 
           - name: cloud-server-discovery-resource-observable-access-in-RFOTM-rep-realloc
             # same configuration as "cloud-server-discovery-resource-observable-access-in-RFOTM-rep-realloc" in the SonarCloud scan job, skip for events that trigger both jobs
             skip: ${{ github.event_name != 'workflow_dispatch' }}
-            args: "-DOC_DISCOVERY_RESOURCE_OBSERVABLE_ENABLED=ON -DOC_RESOURCE_ACCESS_IN_RFOTM_ENABLED=ON -DOC_REPRESENTATION_REALLOC_ENCODING_ENABLED=ON"
+            args: "-DOC_DEBUG_ENABLED=ON -DOC_DISCOVERY_RESOURCE_OBSERVABLE_ENABLED=ON -DOC_RESOURCE_ACCESS_IN_RFOTM_ENABLED=ON -DOC_REPRESENTATION_REALLOC_ENCODING_ENABLED=ON"
           - name: cloud-server-discovery-resource-observable-access-in-RFOTM-rep-realloc-asan
-            args: "-DOC_DISCOVERY_RESOURCE_OBSERVABLE_ENABLED=ON -DOC_RESOURCE_ACCESS_IN_RFOTM_ENABLED=ON -DOC_REPRESENTATION_REALLOC_ENCODING_ENABLED=ON-DOC_ASAN_ENABLED=ON"
+            args: "-DOC_DEBUG_ENABLED=ON -DOC_DISCOVERY_RESOURCE_OBSERVABLE_ENABLED=ON -DOC_RESOURCE_ACCESS_IN_RFOTM_ENABLED=ON -DOC_REPRESENTATION_REALLOC_ENCODING_ENABLED=ON -DOC_ASAN_ENABLED=ON"
           - name: cloud-server-discovery-resource-observable-access-in-RFOTM-rep-realloc-tsan
-            args: "-DOC_DISCOVERY_RESOURCE_OBSERVABLE_ENABLED=ON -DOC_RESOURCE_ACCESS_IN_RFOTM_ENABLED=ON -DOC_REPRESENTATION_REALLOC_ENCODING_ENABLED=ON -DOC_TSAN_ENABLED=ON"
-            clang: true
+            args: "-DOC_DEBUG_ENABLED=ON -DOC_DISCOVERY_RESOURCE_OBSERVABLE_ENABLED=ON -DOC_RESOURCE_ACCESS_IN_RFOTM_ENABLED=ON -DOC_REPRESENTATION_REALLOC_ENCODING_ENABLED=ON -DOC_TSAN_ENABLED=ON"
+            docker_file: docker/apps/Dockerfile.cloud-server-debug-clang
+
+          - name: cloud-server-release-discovery-resource-observable-access-in-RFOTM-rep-realloc
+            docker_file: docker/apps/Dockerfile.cloud-server
+            args: "-DOC_DISCOVERY_RESOURCE_OBSERVABLE_ENABLED=ON -DOC_RESOURCE_ACCESS_IN_RFOTM_ENABLED=ON -DOC_REPRESENTATION_REALLOC_ENCODING_ENABLED=ON"
```

Comment:
Why not just add "-DOC_DEBUG_ENABLED=OFF" here instead of adding "-DOC_DEBUG_ENABLED=ON" everywhere but here?